hockey-score/src/app/teams/team/team.component.spec.ts

```diff
@@ -1,25 +1,100 @@
 import { async, ComponentFixture, TestBed } from '@angular/core/testing';
 
 import { TeamComponent } from './team.component';
+import { NO_ERRORS_SCHEMA, CUSTOM_ELEMENTS_SCHEMA, Component } from '@angular/core';
+import { CommonModule } from '@angular/common';
+import { SharedModule } from 'src/app/shared/shared.module';
+import { MatCardModule, MatButtonModule, MatIconModule } from '@angular/material';
+import { Team } from 'src/app/shared/model/team/team';
+import { Position } from 'src/app/shared/model/position.enum';
+import { Shoots } from 'src/app/shared/model/shoots.enum';
+import { Country } from 'src/app/shared/model/country.enum';
+import { Player } from 'src/app/shared/model/player/player';
+import { By } from '@angular/platform-browser';
 
 describe('TeamComponent', () => {
-  let component: TeamComponent;
-  let fixture: ComponentFixture<TeamComponent>;
+  let component: TestHostComponent;
+  let fixture: ComponentFixture<TestHostComponent>;
 
   beforeEach(async(() => {
     TestBed.configureTestingModule({
-      declarations: [ TeamComponent ]
+      declarations: [TeamComponent, TestHostComponent],
+      schemas: [NO_ERRORS_SCHEMA, CUSTOM_ELEMENTS_SCHEMA],
+      imports: [CommonModule,
+        SharedModule,
+        MatCardModule, MatButtonModule, MatIconModule
+      ]
     })
-    .compileComponents();
+      .compileComponents();
   }));
 
   beforeEach(() => {
-    fixture = TestBed.createComponent(TeamComponent);
+    fixture = TestBed.createComponent(TestHostComponent);
     component = fixture.componentInstance;
     fixture.detectChanges();
   });
 
-  it('should create', () => {
+  it('should create and should bind team item to component', () => {
     expect(component).toBeTruthy();
+    const expectedTeam = component.team;
+    const titleElement = fixture.nativeElement.querySelector('.mat-card-title');
+    const subtitleElement = fixture.nativeElement.querySelector('.mat-card-subtitle');
+    expect(titleElement.textContent).toEqual(expectedTeam.name);
+    expect(subtitleElement.textContent).toEqual(expectedTeam.city);
+  });
+
+  it('should bind team component deleted name event', () => {
+    const deleteButton = fixture.debugElement.query(By.css('.delete-button'));
+    deleteButton.triggerEventHandler('click', null);
+    fixture.detectChanges();
```

Comment:
here we've got to test two main features: input and output

we used TestHostComponent(parent) for testing Input but it can't be used to test inner functionality of TeamComponent (child)

here is it how it can be implemented:

import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { TeamComponent } from './team.component';
import { NO_ERRORS_SCHEMA, CUSTOM_ELEMENTS_SCHEMA, Component } from '@angular/core';
import { CommonModule } from '@angular/common';
import { SharedModule } from 'src/app/shared/shared.module';
import { MatCardModule, MatButtonModule, MatIconModule } from '@angular/material';
import { Team } from 'src/app/shared/model/team/team';
import { Position } from 'src/app/shared/model/position.enum';
import { Shoots } from 'src/app/shared/model/shoots.enum';
import { Country } from 'src/app/shared/model/country.enum';
import { Player } from 'src/app/shared/model/player/player';
import { By } from '@angular/platform-browser';

describe('TeamComponent', () => {
  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [TeamComponent, TestHostComponent],
      schemas: [NO_ERRORS_SCHEMA, CUSTOM_ELEMENTS_SCHEMA],
      imports: [CommonModule,
        SharedModule,
        MatCardModule, MatButtonModule, MatIconModule
      ]
    })
      .compileComponents();
  }));

  describe('TestHostComponent', () => {
    let componentHost: TestHostComponent;
    let fixtureHost: ComponentFixture<TestHostComponent>;

    beforeEach(() => {
      fixtureHost = TestBed.createComponent(TestHostComponent);
      componentHost = fixtureHost.componentInstance;
      fixtureHost.detectChanges();
    });

    it('should create and should bind team item to component', () => {
      expect(componentHost).toBeTruthy();
      const expectedTeam = componentHost.team;
      const titleElement = fixtureHost.nativeElement.querySelector('.mat-card-title');
      const subtitleElement = fixtureHost.nativeElement.querySelector('.mat-card-subtitle');
      expect(titleElement.textContent).toEqual(expectedTeam.name);
      expect(subtitleElement.textContent).toEqual(expectedTeam.city);
    });
  });


  describe('TeamComponent', () => {
    let component: TeamComponent;
    let fixture: ComponentFixture<TeamComponent>;


    beforeEach(() => {
      fixture = TestBed.createComponent(TeamComponent);
      component = fixture.componentInstance;

      component.team = {
        id: 1,
        name: "1stTeam",
        players: [{
          id: 1,
          name: "1stPlayer",
          position: Position.defender,
          number: 78,
          born: new Date(),
          height: 180,
          weight: 90,
          age: 36,
          shoots: Shoots.left,
          country: Country.by,
          stats: []
        }],
        score: 12,
        city: "Hrodna",
        description: "The oldest team"
      };

      fixture.detectChanges();
    });

    it('should bind team component deleted name event', () => {
      spyOn(component.onDelete, 'emit');
      const deleteButton = fixture.debugElement.query(By.css('#deleteBtn'));
      console.warn(deleteButton)
      deleteButton.triggerEventHandler('click', null);

      fixture.detectChanges();


      expect(component.onDelete.emit).toHaveBeenCalledWith(component.team.name);

    });

  });
});


@Component({
  template: `<div> 
    <mat-card class="team-card">
    <mat-card-header>
      <div mat-card-avatar class="team-header-image"></div>
      <mat-card-title>{{team.name}}</mat-card-title>
      <mat-card-subtitle>{{team.city}}</mat-card-subtitle>
    </mat-card-header>
    </mat-card>
    <app-player *ngFor="let player of team.players" [player]="player"></app-player>   
   <button class="delete-button" [team]="team" (deletedName) = "onDelete($event)"></button>
    </div>`
})
class TestHostComponent {
  public team: Team = {
    id: 1,
    name: "1stTeam",
    players: this.players(),
    score: 12,
    city: "Hrodna",
    description: "The oldest team"
  };

  public players(): Player[] {
    return [{
      id: 1,
      name: "1stPlayer",
      position: Position.defender,
      number: 78,
      born: new Date(),
      height: 180,
      weight: 90,
      age: 36,
      shoots: Shoots.left,
      country: Country.by,
      stats: []
    }]
  };
  public deletedName: string;

  public onDelete(team: Team): void {
    this.deletedName = team.name;
  }
}

Reply:
you should also add for delete button in the team.comonent.html

<button mat-icon-button class="mat-icon-button-card-actions" id="deleteBtn" (click)="onDeleteSelect(team)">